### DISCUSSION TITLE
Match type deduction does not work for union types - bug or feature?

### DISCUSSION THREAD

**Discussion by scf37** [scala/scala3#19989] [General Question]
I was hoping v type to be deduced as `E.B | E.C` but it seems to be just `E`
```scala
enum E:
  case A()
  case B()
  case C()

def foo(e: E): E.B | E.C  = e match  
  case _: E.A => ???
  case v => v // Found:    (v : E)  Required: E.B | E.C
```

**Comment by bishabosha** [comment]
Well it has always been the case that a variable pattern is exactly the same type as the scrutinee, it might break currently compiling code to infer more precise types
